scipy/optimize/_hessian_update_strategy.py

```diff
@@ -254,10 +290,13 @@ class BFGS(FullHessianUpdateStrategy):
         unaffected by the exception strategy. By default is equal to
         1e-8 when ``exception_strategy = 'skip_update'`` and equal
         to 0.2 when ``exception_strategy = 'damp_update'``.
-    init_scale : {float, 'auto'}
-        Matrix scale at first iteration. At the first
-        iteration the Hessian matrix or its inverse will be initialized
-        with ``init_scale*np.eye(n)``, where ``n`` is the problem dimension.
+    init_scale : {float, np.array, 'auto'}
+        At the first iteration the Hessian matrix or its inverse will
+        be initialized with the scale. In case the dimensionality of
+        the scale is ``(n, n)``, the scale itself will be used as
+        the initial matrix.
+        Otherwise the matrix will be initialized with
+        ``np.eye(n) * init_scale``, where ``n`` is the problem dimension.
```

Comment:
I think we can be a bit more expressive. I think if I understand correctly, an np.array without (n x n) shape is discarded silently. Hence should not it be resulting in an error instead? 

```python
"""
        This parameter can be used to initialize the Hessian or its
        inverse. When a float is given, the relevant array is initialized
        to ``np.eye(n) * init_scale``, where ``n`` is the problem dimension.
        Alternatively, if a precisely ``(n, n)`` shaped array is given, this 
        array will be used. Otherwise an error is generated.
        Set it to 'auto' in order to use an automatic heuristic for choosing
        the initial scale. The heuristic is described in [1]_, p.143.
        The default is 'auto'.
"""
```

Reply:
Yes, that's nicely formulated, I added the "symmetric" requirement.

> an np.array without (n x n) shape is discarded silently. 

An error is raised already. Either a "scalar-like" (`np.size(init) == 1`) or an array with shape (n, n), symmetric, is allowed. Everything else raises an error.